plugins/kafka/kafka_input.go

```diff
@@ -170,9 +170,7 @@ func (k *KafkaInput) Init(config interface{}) (err error) {
 			if err = os.MkdirAll(filepath.Dir(k.checkpointFilename), 0766); err != nil {
 				return
 			}
-			if err = k.writeCheckpoint(0); err != nil {
-				return
-			}
+			k.consumerConfig.OffsetMethod = sarama.OffsetMethodOldest
```

Comment:
We want to start from the oldest point but only the first time we don't have a checkpoint.  Once we start from there we want to keep track of where we are.  This change will always process the full history every time Heka is restarted.